yt_dlp/extractor/kanal2.py

```diff
@@ -0,0 +1,122 @@
+# coding: utf-8
+from __future__ import unicode_literals
+from datetime import datetime
+import re
+import time
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    url_or_none,
+    traverse_obj,
+)
+
+
+class Kanal2IE(InfoExtractor):
+    SUBTITLE_DATE_RE = re.compile(r'\((\d{2}\.\d{2}\.\d{4}\s\d{2}:\d{2})\)$')
+
+    _VALID_URL = r'https?://.+\.postimees\.ee/[^?]+\?(.*?&)?id=(?P<id>\d+)'
+    _TESTS = [
+        {
+            'note': 'Test standard url (#5575)',
+            'url': 'https://kanal2.postimees.ee/pluss/video/?id=40792',
+            'md5': 'baf49cab468cf553bc1cf36909f1a0c7',
+            'info_dict': {
+                'id': '40792',
+                'ext': 'mp4',
+                'title': 'Aedniku aabits / Osa 53  (05.08.2016 20:00)',
+                'thumbnail': r're:https?://.*\.jpg$',
+                'description': 'md5:53cabf3c5d73150d594747f727431248',
+                'upload_date': '20160805',
+                'timestamp': 1470416400,
+            }
+        },
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        playlist = self.get_playlist(video_id)
+
+        info = {
+            'id': video_id,
+            'title': self.get_title(playlist.get('info')),
+            'description': traverse_obj(playlist, ('info', 'description')),
+            'webpage_url': traverse_obj(playlist, ('data', 'url')),
+            'thumbnail': traverse_obj(playlist, ('data', 'image')),
+            'formats': self.get_formats(playlist, video_id),
+            'timestamp': self.get_timestamp(traverse_obj(playlist, ('info', 'subtitle'))),
+        }
+
+        return info
+
+    def get_title(self, info):
+        title = info['title']
+
+        if info['subtitle']:
+            title += ' / ' + info['subtitle']
+
+        return title
+
+    def get_timestamp(self, subtitle):
+        if not subtitle:
+            return None
+        # Extract timestamp from:
+        #  "subtitle": "Osa 53  (05.08.2016 20:00)",
+        match = self._search_regex(self.SUBTITLE_DATE_RE, subtitle, 'dateandtime', default=None)
+        if not match:
+            return None
+
+        # https://stackoverflow.com/a/27914405/2314626
+        date = datetime.strptime(match, '%d.%m.%Y %H:%M')
+        unixtime = time.mktime(date.timetuple())
+
+        try:
+            import pytz
+            # the time is always in this timezone
+            tz = pytz.timezone('Europe/Tallinn')
+            unixtime += tz.utcoffset(date).microseconds
+        except ImportError:
+            pass
+
+        return unixtime
```

Comment:
one-time use function; use `unified_timestamp` instead, could even be inlined in return:
```py
            'timestamp': unified_timestamp(
                self._search_regex(
                    r'\((\d{2}\.\d{2}\.\d{4}\s\d{2}:\d{2})\)$', sub,
                    'dateandtime', default=None))
```

Reply:
the string that is extracted with regex is always Europe/Tallinn timezone, regardless what is the timezone of the user computer. how do enforce that?

Reply:
also `"airdate": "20160805"` doesn't include time.

Reply:
```
AssertionError: 1470416400 != 1470427200 : Invalid value for field timestamp, expected 1470416400, got 1470427200

----------------------------------------------------------------------
Ran 1 test in 1.928s

FAILED (failures=1)
➜ TZ=Europe/Tallinn utime 1470416400
Fri Aug  5 20:00:00 2016 +0300
➜ TZ=Europe/Tallinn utime 1470427200
Fri Aug  5 23:00:00 2016 +0300
```

Reply:
```py
            'timestamp': unified_timestamp(
                self._search_regex(
                    r'\((\d{2}\.\d{2}\.\d{4}\s\d{2}:\d{2})\)$', sub,
                    'dateandtime', default='') + ' +0200')
```

Reply:
`' +0200'` is not correct half of the year due DST changes. and if you know timezones, they are political decisions, and can't be calculated.

Reply:
I got the `timestamp` returned from `unified_timestamp()` corrected with this snippet:

```py
    def patch_timezone(self, timestamp: int):
        """
        unified_timestamp() assumed date is in UTC, now let's fix the time using pytz data
        """
        import datetime
        
        date = datetime.datetime.fromtimestamp(timestamp, datetime.timezone.utc)
        t = time.mktime(date.timetuple())
```

but this will only work if local timezone is `Europe/Tallinn`. To get it work universally need to load `Europe/Tallinn` database from [pytz](https://pypi.org/project/pytz) package.

```
TZ=Europe/Tallinn python test/test_download.py TestDownload.test_Kanal2 # works
TZ=Europe/London python test/test_download.py TestDownload.test_Kanal2  # fails
```

Reply:
where is this timestamp used anyway? as filename is created from `title` i think. how precise it needs to be? i personally prefer to be most accurate, but may need to downsize due extra pytz dependency. can the extractor use optional dependency?

there are some deps in requirements.txt, pytz needs to be added there too. are those deps bundled when the release dist is created?

Reply:
with da22e7c76 passes:

```
➜ TZ=Europe/London python test/test_download.py TestDownload.test_Kanal2
```

Reply:
> where is this timestamp used anyway? as filename is created from title i think. how precise it needs to be? 

`timestamp` is an info_dict field, just like `title`, or `id`, or `description`. It is metadata about the video that has a number of applications, e.g. embedding it, writing it to an info.json file, using it in the output template (`%(title)s [%(id)s].%(ext)s` is the default, but it can be any combination you want it to be).

The value of `timestamp` should be as accurate as possible *within reason*. For some sites, such as Youtube, there is no exact time available at all, only an `upload_date`. IMO introducing a new dependency to improve the accuracy of one extractor's timestamp by 1 hour for 6 months out of the year is **NOT** within reason. 

pukkandan's suggestion above is a perfectly reasonable solution, and will result in timestamps that are accurate *enough*.

Reply:
Removed b4673172e17811d630f3cfae14e82f7dd96ac444, b4c61af8e940e4b85d3677ddc22c3196c212348b commits

Reply:
what about adding code that if downloading with the local timezone Europe/Tallinn, the value would be accurate? currently, it's always incorrect. this snippet would do it:
- https://github.com/yt-dlp/yt-dlp/pull/5575#discussion_r1027293523

Reply:
we don't want the values in the tests to vary depending on where the tests are being run

Reply:
good point. but I was thinking more about user experience. but since the two are closely related I'll update the test data with the current "truth" then.

Reply:
Done with 6036b1c44

Reply:
I think https://github.com/yt-dlp/yt-dlp/pull/5575#discussion_r1027284827 is a good enough solution. But in either case, the test definition should be the correct value

Reply:
so, GMT+2 it is then: 2d5176fe2